app/src/main/java/com/example/footballmanager/screens/view_models/HomeViewModel.kt

```diff
@@ -8,33 +8,52 @@ import androidx.compose.runtime.setValue
 import androidx.core.content.ContextCompat.getString
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.example.footballmanager.network.structures.FixtureDataWrapper
 import com.example.footballmanager.R
-import com.example.footballmanager.network.FixtureDataWrapper
 import com.example.footballmanager.network.FootballApi
+import com.example.footballmanager.network.FootballApiService
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import java.io.IOException
 import java.time.LocalDate
 import java.time.format.DateTimeFormatter
 import java.time.format.TextStyle
 import java.util.Locale
+import javax.inject.Inject
 
 const val DAYS_OFFSET = 50
 
-class HomeViewModel : ViewModel() {
+@HiltViewModel
+class HomeViewModel @Inject constructor(
+    private val footballApiService: FootballApiService
+) : ViewModel() {
 
-    var retrievingDataState: RetrievingDataState by mutableStateOf(RetrievingDataState.Loading)
+    var retrievingByDateState: RetrievingDataState by mutableStateOf(RetrievingDataState.Loading)
         private set
+    var retrievingByLiveNowState: FixtureDataWrapper by mutableStateOf(FixtureDataWrapper())
+        private set
+
+    fun getFixturesLiveNow() {
+        viewModelScope.launch {
+            runCatching {
+                val result = footballApiService.getFixturesByLiveNow()
+                retrievingByLiveNowState = result
+            }
+        }
+    }
 
-    fun getFixturesData(date: String = LocalDate.now().toString(), ctx:Context) {
+    fun getFixturesData(date: String = LocalDate.now().toString(), ctx: Context) {
         viewModelScope.launch {
             try {
-                val result = FootballApi.retrofitService.getFixtures(date = date)
+                val result = footballApiService.getFixturesByDate(date = date)
```

Comment:
if responseBody is array you can use `isEmpty()` method. IMO is better but this also is OK